### ISSUE TITLE
AIOB generated workload for NS3 simulator, TP missing backward communication operation

### ISSUE THREAD

**Issue by WWeiOne** [aliyun/SimAI#123]
During backpropagation, the input gradients typically require a reduce or gather operation. However, I noticed that the backward pass for TP comm does not include any communication operations. Could you please help clarify the reason for this?

`sh ./scripts/megatron_workload_with_aiob.sh -m 175 --aiob_enable --comp_filepath workload/aiob_inputs/Example.txt`

generated
<img width="664" alt="Image" src="https://github.com/user-attachments/assets/138a803f-ccd2-4117-a75f-329d8a06a9ce" />

tutorial
![Image](https://github.com/user-attachments/assets/2b750ad7-3fb9-4a7a-8597-616ff29835b2)

**Comment by tianhao909** [comment]
> During backpropagation, the input gradients typically require a reduce or gather operation. However, I noticed that the backward pass for TP comm does not include any communication operations. Could you please help clarify the reason for this?
> 
> `sh ./scripts/megatron_workload_with_aiob.sh -m 175 --aiob_enable --comp_filepath workload/aiob_inputs/Example.txt`
> 
> generated <img alt="Image" width="664" src="https://private-user-images.githubusercontent.com/109638642/439791371-138a803f-ccd2-4117-a75f-329d8a06a9ce.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDczMTM2MTIsIm5iZiI6MTc0NzMxMzMxMiwicGF0aCI6Ii8xMDk2Mzg2NDIvNDM5NzkxMzcxLTEzOGE4MDNmLWNjZDItNDExNy1hNzVmLTMyOWQ4YTA2YTljZS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTE1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUxNVQxMjQ4MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iYzYzMDc2ZjU2NDE2OWE0YjA0ZWQ4NTk4MWM2N2Y2NDdkMGExMTUzOGJhOTAyMTQzOWEzMjQ3ODg0YmQyZTA0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.2nd4quq8YGT-TIA9Y81C6Sme2Azb2iU9f3Ww3mx2qcQ">
> 
> tutorial ![Image](https://github.com/user-attachments/assets/2b750ad7-3fb9-4a7a-8597-616ff29835b2)

"In LLM training, the backward AllReduce communication can be overlapped with the computation kernel of the backward pass
